source/ncengine/graphics2/diligent/pass/PassUtilities.cpp

```diff
@@ -132,6 +132,27 @@ void BindRenderTarget(Diligent::IDeviceContext& context,
     context.SetRenderTargets(colorRenderTargetIndex == NoTarget ? 0 : 1, &pRTV, pDSV, Diligent::RESOURCE_STATE_TRANSITION_MODE::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
 }
 
+auto CreateShaderFromSourceIfInitialized(ShaderFactory& shaderFactory, Diligent::SHADER_TYPE shaderType, const nc::graphics::ShaderPaths& shaderPaths) -> Diligent::RefCntAutoPtr<Diligent::IShader>
```

Comment:
I don't love that this helper function also handles the empty checking/null construction. To me, its better to express that directly in the PSO construction, if nullness is valid for that pipeline. I can live with it, though.

Reply:
That's really the only thing it does - Nullness is valid for the pipeline. I think the right move is to consolidate all the pipeline creation state into a method as we have a lot of duplicate code everywhere. I'll probably do that after we move to deferred because much will change